src/core/thread/mle.cpp

```diff
@@ -1792,6 +1792,8 @@ ThreadError Mle::HandleParentResponse(const Message &aMessage, const Ip6::Messag
 
     otLogInfoMle("Received Parent Response\n");
 
+    assert(threadMessageInfo != NULL);
```

Comment:
I'm still seeing this in `Mle::HandleParentResponse()`.  Why not put this in `Mle::HandleUdpReceive()` as I suggested above?